src/main/java/org/zalando/nakadi/metrics/MonitoringRequestFilter.java

```diff
@@ -3,61 +3,56 @@
 
     private final Timer httpConnectionsTimer;
     private final Counter openHttpConnectionsCounter;
     private final MetricRegistry perPathMetricRegistry;
 
-    public MonitoringRequestFilter(final MetricRegistry overallMetricRegistry,
-                                   final MetricRegistry perPathMetricRegistry) {
-        openHttpConnectionsCounter = overallMetricRegistry.counter(MetricUtils.NAKADI_PREFIX
+    @Autowired
+    public MonitoringRequestFilter(final MetricRegistry metricRegistry,
+                                   @Qualifier("perPathMetricRegistry") final MetricRegistry perPathMetricRegistry) {
+        openHttpConnectionsCounter = metricRegistry.counter(MetricUtils.NAKADI_PREFIX
                 + "general.openSynchronousHttpConnections");
-        httpConnectionsTimer = overallMetricRegistry
+        httpConnectionsTimer = metricRegistry
                 .timer(MetricUtils.NAKADI_PREFIX + "general.synchronousHttpConnections");
         this.perPathMetricRegistry = perPathMetricRegistry;
     }
 
     @Override
-    public void init(final FilterConfig filterConfig) throws ServletException {
-        // nothing to do
-    }
-
-    @Override
-    public void doFilter(final ServletRequest request, final ServletResponse response, final FilterChain chain)
+    protected void doFilterInternal(final HttpServletRequest request,
+                                    final HttpServletResponse response, final FilterChain filterChain)
             throws IOException, ServletException {
         openHttpConnectionsCounter.inc();
         final Timer.Context timerContext = httpConnectionsTimer.time();
 
-        Timer.Context perPathTimerContext = null;
-        if (request instanceof HttpServletRequest) {
-            final HttpServletRequest httpRequest = (HttpServletRequest) request;
-            final String clientId = Optional.ofNullable(httpRequest.getUserPrincipal())
-                    .map(Principal::getName)
-                    .orElse("unauthenticated");
-            final String perPathMetricKey = MetricRegistry.name(
-                    clientId,
-                    httpRequest.getMethod(),
-                    httpRequest.getServletPath());
+        final String clientId = Optional.ofNullable(request.getUserPrincipal())
+                .map(Principal::getName)
+                .orElse("unauthenticated");
+        final String perPathMetricKey = MetricRegistry.name(
+                clientId,
+                request.getMethod(),
+                request.getServletPath());
 
-            perPathTimerContext = perPathMetricRegistry.timer(perPathMetricKey).time();
-        }
+        final Timer.Context perPathTimerContext = perPathMetricRegistry.timer(perPathMetricKey).time();
 
         try {
 
-            chain.doFilter(request, response);
+            filterChain.doFilter(request, response);
 
         } finally {
             if (perPathTimerContext != null) {
```

Comment:
Not at all. I forgot to remove it after refactoring. Done https://github.com/zalando/nakadi/pull/578/commits/c06e4871d24dd3e67f4a6d41f22f5b01ab27b1b7